api/src/routes/donate.ts

```diff
@@ -214,3 +197,119 @@ export const donateRoutes: FastifyPluginCallbackTypebox = (
 
   done();
 };
+
+/**
+ * Plugin for the donation endpoints.
+ *
+ * @param fastify The Fastify instance.
+ * @param _options Options passed to the plugin via `fastify.register(plugin, options)`.
+ * @param done The callback to signal that the plugin is ready.
+ */
+export const chargeStripeRoute: FastifyPluginCallbackTypebox = (
+  fastify,
+  _options,
+  done
+) => {
+  // Stripe plugin
+  const stripe = new Stripe(STRIPE_SECRET_KEY, {
+    apiVersion: '2020-08-27',
+    typescript: true
+  });
+
+  // @ts-expect-error - @fastify/csrf-protection needs to update their types
+  // eslint-disable-next-line @typescript-eslint/unbound-method
+  fastify.addHook('onRequest', fastify.csrfProtection);
+  fastify.addHook('onRequest', fastify.addUserIfAuthorized);
+  fastify.post(
+    '/donate/charge-stripe',
+    {
+      schema: schemas.chargeStripe
+    },
+    async (req, reply) => {
+      try {
+        const id = req.user?.id;
+        const { email, name, token, amount, duration } = req.body;
+
+        // verify the parameters
+        if (
+          !isEmail(email) ||
+          !donationSubscriptionConfig.plans[duration].includes(amount)
+        ) {
+          void reply.code(500);
+          return {
+            error: 'The donation form had invalid values for this submission.'
+          } as const;
+        }
+
+        // TODO(Post-MVP) new users should not be created if user is not found
+        const user = id
+          ? await fastify.prisma.user.findUniqueOrThrow({ where: { id } })
+          : await findOrCreateUser(fastify, email);
+
+        const { id: customerId } = await stripe.customers.create({
+          email,
+          name
+        });
+
```

Comment:
We should figure out if this might benefit from being done **before** creating the customer.

```suggestion

        const { id: paymentMethodId } = await stripe.paymentMethods.create({
          type: 'card',
          card: { token: token.id }
        });

```

Reply:
This looks to be outdated.